src/apple/disk.rs

```diff
@@ -53,14 +62,338 @@ impl DiskExt for Disk {
+type RetainedCFURL = CFReleaser<core_foundation_sys::url::__CFURL>;
+
+unsafe fn build_requested_properties(properties: &[CFStringRef]) -> Option<RetainedCFArray> {
+    CFReleaser::new(CFArrayCreate(
+        ptr::null_mut(),
+        properties.as_ptr() as *const *const c_void,
+        properties.len() as CFIndex,
+        &core_foundation_sys::array::kCFTypeArrayCallBacks,
+    ))
+}
+
+fn get_disk_properties(
+    volume_url: &RetainedCFURL,
+    requested_properties: &RetainedCFArray,
+) -> Option<RetainedCFDictionary> {
+    CFReleaser::new(unsafe {
+        ffi::CFURLCopyResourcePropertiesForKeys(
+            volume_url.inner(),
+            requested_properties.inner(),
+            ptr::null_mut(),
+        )
+    })
+}
+
+fn get_available_volume_space(disk_props: &RetainedCFDictionary) -> u64 {
+    // We prefer `AvailableCapacityForImportantUsage` over `AvailableCapacity` because
+    // it takes more of the system's properties into account, like the trash, system-managed caches,
+    // etc. It generally also returns higher values too, because of the above, so its a more accurate
+    // representation of what the system _could_ still use.
+    unsafe {
+        get_int_value(
+            disk_props.inner(),
+            DictKey::Extern(ffi::kCFURLVolumeAvailableCapacityForImportantUsageKey),
+        )
+        .filter(|bytes| *bytes != 0)
+        .or_else(|| {
+            get_int_value(
+                disk_props.inner(),
+                DictKey::Extern(ffi::kCFURLVolumeAvailableCapacityKey),
+            )
+        })
+    }
+    .unwrap_or_default() as u64
+}
+
+pub(super) enum DictKey {
+    Extern(CFStringRef),
+    #[cfg(target_os = "macos")]
+    Defined(&'static str),
+}
+
+unsafe fn get_dict_value<T, F: FnOnce(*const c_void) -> Option<T>>(
+    dict: CFDictionaryRef,
+    key: DictKey,
+    callback: F,
+) -> Option<T> {
+    #[cfg(target_os = "macos")]
+    let _defined;
+    let key = match key {
+        DictKey::Extern(val) => val,
+        #[cfg(target_os = "macos")]
+        DictKey::Defined(val) => {
+            _defined = CFReleaser::new_unchecked(cfs::CFStringCreateWithBytesNoCopy(
+                kCFAllocatorDefault,
+                val.as_ptr(),
+                val.len() as isize,
+                cfs::kCFStringEncodingUTF8,
+                false as _,
+                core_foundation_sys::base::kCFAllocatorNull,
+            ));
+
+            _defined.inner()
+        }
+    };
+
+    let mut value = std::ptr::null();
+    if CFDictionaryGetValueIfPresent(dict, key.cast(), &mut value) != 0 {
+        callback(value)
+    } else {
+        None
+    }
+}
+
+pub(super) unsafe fn get_str_value(dict: CFDictionaryRef, key: DictKey) -> Option<String> {
+    get_dict_value(dict, key, |v| {
+        let v = v as cfs::CFStringRef;
+
+        let len_utf16 = cfs::CFStringGetLength(v);
+        let len_bytes = len_utf16 as usize * 2; // Two bytes per UTF-16 codepoint.
```

Comment:
```suggestion
        let len_utf16: usize = cfs::CFStringGetLength(v) as _;
        let len_bytes = len_utf16 * 2; // Two bytes per UTF-16 codepoint.
```

Reply:
I've cleaned this one up too. It now does `as usize` when binding `len_utf16`.